test/constructors.jl

```diff
@@ -28,25 +28,72 @@ module TestConstructors
     @test isequal(dt[:x1], NullableArray(dt2[:x1]))
     @test isequal(dt[:x2], NullableArray(dt2[:x2]))
 
-    @test isequal(dt, DataTable(x1 = [0.0, 0.0, 0.0],
-                                x2 = [1.0, 1.0, 1.0]))
-    @test isequal(dt, DataTable(x1 = [0.0, 0.0, 0.0],
-                                x2 = [1.0, 1.0, 1.0],
+    @test isequal(dt, DataTable(x1 = NullableCategoricalVector([0.0, 0.0, 0.0]),
+                                x2 = NullableCategoricalVector([1.0, 1.0, 1.0])))
+    @test isequal(dt, DataTable(x1 = NullableCategoricalVector([0.0, 0.0, 0.0]),
+                                x2 = NullableCategoricalVector([1.0, 1.0, 1.0]),
                                 x3 = [2.0, 2.0, 2.0])[[:x1, :x2]])
 
     dt = DataTable(Int, 2, 2)
     @test size(dt) == (2, 2)
-    @test eltypes(dt) == [Nullable{Int}, Nullable{Int}]
+    @test eltypes(dt) == [Int, Int]
 
     dt = DataTable([Int, Float64], [:x1, :x2], 2)
     @test size(dt) == (2, 2)
-    @test eltypes(dt) == [Nullable{Int}, Nullable{Float64}]
-
-    @test isequal(dt, DataTable([Int, Float64], 2))
```

Comment:
OK. So use `Nullable` eltypes, which should compare as `isequal` even with nulls values.